src/commands/pull.js

```diff
@@ -0,0 +1,98 @@
+
+        const { actorId } = args;
+        const actor = await apifyClient.actor(actorId).get();
+        if (!actor) throw new Error(`Cannot find actor with ID '${actorId}' in your account.`);
+        const { name, versions } = actor;
+
+        const latestVersion = versions.reduce((match, curr) => {
+            if (semverGt(`${curr.versionNumber}.0`, `${match.versionNumber}.0`)) return curr;
+            return match;
+        });
+
+        const dirpath = path.join(cwd, name);
+        mkdirp.sync(dirpath);
+
+        if (!(fs.readdirSync(dirpath).length === 0)) {
+            error(`Directory ${dirpath} is not empty. Please empty it or choose another directory.`);
+        }
+
+        let isPullSuccessful = false;
+
+        switch (latestVersion.sourceType) {
+            case 'TARBALL': {
+                const { tarballUrl } = latestVersion;
+
+                const { data } = await get(tarballUrl, { responseType: 'arraybuffer' });
+
+                const zipFile = new AdmZip(Buffer.from(data, 'binary'));
+                zipFile.extractAllTo(dirpath);
+                break;
+            }
+            case 'SOURCE_FILES': {
+                const { sourceFiles } = latestVersion;
+                for (const file of sourceFiles) {
+                    if (!file.folder) {
+                        mkdirp.sync(dirpath);
+                        fs.writeFileSync(`${dirpath}/${file.name}`, file.content);
+                    }
+                }
+                isPullSuccessful = true;
+                break;
+            }
+            case 'GIT_REPO': {
+                // e.g. https://github.com/jakubbalada/Datasety.git#master:RejstrikPolitickychStran
+                const { gitRepoUrl } = latestVersion;
+                const [repoUrl, branchDirPart] = gitRepoUrl.split('#');
+
+                let branch;
+                let dir;
+                if (branchDirPart) [branch, dir] = branchDirPart.split(':');
+
+                const emitter = tiged(branchDirPart ? `${repoUrl}/${dir}#${branch}` : repoUrl);
+                try {
+                    await emitter.clone(dirpath);
+                } catch (err) {
+                    error(`Failed to pull actor from ${gitRepoUrl}.`);
+                    break;
+                }
+
+                isPullSuccessful = true;
+                break;
+            }
+            case 'GITHUB_GIST':
+                throw new Error('Pulling from GitHub Gist is not supported.');
+            default:
+                throw new Error(`Unknown source type: ${latestVersion.sourceType}`);
+        }
+
+        if (isPullSuccessful) success(`Pulled to ${dirpath}/`);
+    }
+}
+
+PullCommand.description = 'Pulls the latest version of an actor from the Apify platform to the current directory. ';
+
+PullCommand.args = [
+    {
+        name: 'actorId',
+        required: true,
+        description: 'ID or name (username/actor_name) of an existing actor on the Apify platform which will be pulled. ',
```

Comment:
Does it have to be `username/actor_name`? We could really simply support just `name` by requesting `~${name}` from the API.

Reply:
Also, we can adjust this in next iteration. This wasn't required in current spec.

Reply:
The spec can't think about every single small thing that you might encounter during implementation. And it's literally like 1 line of code to support this.